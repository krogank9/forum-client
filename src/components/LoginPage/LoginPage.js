import React, { Component } from 'react';
import { withRouter } from "react-router-dom";

import ForumApiService from '../../services/forum-api-service';
import ForumContext from '../../contexts/ForumContext';

class LoginPage extends Component {
  static contextType = ForumContext

  constructor() {
    super();
    
    this.state = {
      userName: "",
      password: ""
    }
  }

  handleSubmit = (evt) => {
    evt.preventDefault();
    const data = new FormData(evt.target);

    ForumApiService.login(this.state.userName, this.state.password)
      .then(json => {
        console.log(json)
        this.context.onUserLoggedIn(json.authToken, json.userName, json.userId)
        this.props.history.goBack()
      })
      .catch(e => {
        console.log(e)
        alert(`Error logging in: ${e.error}`)
      })
  }

  handleCreateAccount = (evt) => {
    ForumApiService.registerUser(this.state.userName, this.state.password)
      .then(json => {
        console.log(json)
        alert("Account created successfuly")
      })
      .catch(e => {
        console.log(e)
        alert(`Error creating account: ${e.error}`)
      })
  }

  updateFormState = (evt) => {
    let name = evt.target.getAttribute("name")

    this.setState({ [name]: evt.target.value })
  }

  render() {
    return (
      <section>
        <h2>
          Login/Create Account
        </h2>

        <form onSubmit={this.handleSubmit}>
          Username:
          <input type="text" name="userName" onChange={this.updateFormState} />

          <br />

          Password:
          <input type="password" name="password" onChange={this.updateFormState} />

          <br />

          <input type="button" value="Create Account" name="create_account" onClick={this.handleCreateAccount} />

          <input type="submit" value="Log In" name="login" />
        </form>
      </section>
    );
  }
}

export default withRouter(LoginPage);