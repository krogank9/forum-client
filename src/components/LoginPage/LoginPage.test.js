import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from "react-router-dom";
import LoginPage from './LoginPage';

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`LoginPage component`, () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <LoginPage />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });


  it('renders the LoginPage', () => {
    const wrapper = shallow(
      <BrowserRouter>
        <LoginPage/>
      </BrowserRouter>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})