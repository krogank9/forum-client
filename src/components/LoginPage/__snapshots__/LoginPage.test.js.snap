// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders the LoginPage 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(LoginPage) />
</Router>
`;
